### ISSUE TITLE
Got lots of `SSLException` `SSLEngine closed already` error when restart the server

### ISSUE THREAD

**Issue by dozer47528** [netty/netty#7142]
### Expected behavior
Without `SSLException` `SSLEngine closed already` error

### Actual behavior
Got lots of `SSLException` `SSLEngine closed already` error

### Steps to reproduce
Restart the server.
And more than 20k clients will connect to the server at the same time.

### Minimal yet complete reproducer code (or URL to code)
This is a WebSocket server with SSL.

### Netty version
4.1.12.Final

### JVM version (e.g. `java -version`)
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

### OS version (e.g. `uname -a`)
Linux ip-10-2-0-171 4.4.0-1026-aws #35-Ubuntu SMP Thu Jul 20 21:59:09 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

=============================================
Parts of the debug logs:

2017-08-21 09:42:35.700 DEBUG 21882 --- [nioEventLoopGroup-2-1] io.netty.handler.logging.LoggingHandler  : [id: 0x6b3c59d2, L:/0:0:0:0:0:0:0:0:443] READ: [id: 0xd881b5c6, L:/10.0.1.246:443 - R:/124.248.220.19:33047]
2017-08-21 09:42:35.700 DEBUG 21882 --- [nioEventLoopGroup-2-1] io.netty.handler.logging.LoggingHandler  : [id: 0x6b3c59d2, L:/0:0:0:0:0:0:0:0:443] READ COMPLETE
2017-08-21 09:42:47.445 DEBUG 21882 --- [nioEventLoopGroup-3-1] io.netty.handler.ssl.SslHandler          : [id: 0xd881b5c6, L:/10.0.1.246:443 - R:/124.248.220.19:33047] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
2017-08-21 09:42:54.966 DEBUG 21882 --- [nioEventLoopGroup-3-1] c.n.server.handler.AminoAuthHandler      : Channel:[id: 0xd881b5c6, L:/10.0.1.246:443 - R:/124.248.220.19:33047] Uid:58889fcf-30eb-4b9d-9a19-9c0c88fe8297 Login
2017-08-21 09:42:54.966 DEBUG 21882 --- [nioEventLoopGroup-3-1] i.n.h.c.h.w.WebSocketServerHandshaker    : [id: 0xd881b5c6, L:/10.0.1.246:443 - R:/124.248.220.19:33047] WebSocket version V13 server handshake
2017-08-21 09:42:54.966 DEBUG 21882 --- [nioEventLoopGroup-3-1] i.n.h.c.h.w.WebSocketServerHandshaker    : WebSocket version 13 server handshake key: d2Enrby2GG0Zuh7j8kx9Ug==, response: RQb/6x9CjnzhnmIeXdQQ/s0TWKo=
2017-08-21 09:42:54.966 ERROR 21882 --- [nioEventLoopGroup-3-1] c.n.server.handler.ExceptionHandler      : SSLEngine closed already

javax.net.ssl.SSLException: SSLEngine closed already
        at io.netty.handler.ssl.SslHandler.wrap(...)(Unknown Source)

**Comment by dozer47528** [comment]
That's so wired, because the SSL handshake is done, and the server received the WebSocket handshake request from the client.

**Comment by dozer47528** [comment]
Fixed, not a Netty issue.
The JDK SSL engine is very slow, the clients are time out, this is the root cause.

**Comment by ssheng** [comment]
@dozer47528 Could you please how you discover the problem. We're seeing the same error but would like to confirm if they are the same cause.

**Comment by ssheng** [comment]
Earlier I mentioned that we encountered the same problem and let me share our findings here.

We found out that the "javax.net.ssl.SSLException: SSLEngine closed already"is caused by creating the SSLEngine with a host and port.

  sslEngine = sslContext.createSSLEngine(host, port);

If we were to create the SSLEngine without the host and port, the error no longer occurs. The reason we create the SSLEngine with a host and port is that it is supposed to leverage session resumption and has empirically reduced connection establish time.

  sslEngine = sslContext.createSSLEngine();

We did this for half of the instances of a particular service. When the issue reoccured, on those instances that creates the SSLEngine with the host and port saw the SSLException.

Hence, we suspect this is a bug in the JDK. We will upgrade to a later release of Java and retry.
